engine/src/main/java/org/hibernate/validator/internal/util/annotationfactory/AnnotationProxy.java

```diff
@@ -176,6 +182,18 @@ else if ( m.getDefaultValue() != null ) {
 		return result;
 	}
 
+	private Map<String, Object> getAnnotationValues(Annotation annotation) {
+		if ( Proxy.isProxyClass( annotation.getClass() ) ) {
+			InvocationHandler invocationHandler = Proxy.getInvocationHandler( annotation );
+			if ( invocationHandler instanceof AnnotationProxy ) {
```

Comment:
Not sure what is unclear. I replaced `is a proxy` by `is an AnnotationProxy` in the commit message.